rmf_robot_sim_common/src/slotcar_common.cpp

```diff
@@ -987,7 +986,7 @@ void SlotcarCommon::charge_state_cb(
 
 bool SlotcarCommon::near_charger(const Eigen::Isometry3d& pose) const
 {
-  std::string lvl_name = get_level_name(pose.translation()[2]);
+  std::string lvl_name = _last_known_level;
```

Comment:
From what I understand with the implemented change in the `get_level_name` function it should be OK to still use it since it would return `_last_known_level` as long as the robot is not close enough to a level?
It's a very unlikely scenario but IMHO this change could be a bit tricky if in the future people call this function without calling one of the others that calls `get_level_name` such as `publish_state_topic`, then this function would work with outdated level information?

Reply:
Yeah it should be possible but I'm returning the cached value here to reduce computation load. I feel its fine to do this for this function as it is only used to check if the robot is near a charger which shouldn't change when the robot is transitioning between levels. The main update loop is calling `get_level_name(~)` at a pretty high rate anyways.

Reply:
I guess in the current scenario version your code would work, it would be a bit undefined in case things were to be moved around a bit, for example if `publish_robot_state` was to be moved to be called _after_ `near_charger` in the main loop the first simulation would have uninitialized level name, and in general now the level will only be updated at 2 Hz instead of every simulation iteration (but I guess that's also fine for the sake of the charger?)

Reply:
I don't feel too strongly about this so happy to revert to the original function call here.